### PR TITLE
Fixes Publishers page displaying only 2 pages of works #2471 Attempt 2

### LINKED ISSUE #2287: Publishers page displaying only 2 pages of works
### Description
This page says "15 works", but only displays 12

### Relevant url?
- Publisher page: https://openlibrary.org/publishers/39_West_Press
- Searching for the publisher directly (which displays all the works): https://openlibrary.org/search?q=publisher%3A"39+West+Press"&mode=everything

### Expectation
Should display the right number of works.

### Details

- **Logged in (Y/N)?** Y
- **Browser type/version?** Firefox 68.0.1
- **Operating system?** Windows 10

### PR DESCRIPTION
**Pull request by cbolgiano**
Closes #2287 

### Technical
In `Carousel.js` we now check to make sure that we continue to fetch until we are truly on the last page.  Changed the default `page.mode` of the `custom_carousel` to `page` instead of `offset`.

### Testing
1. Login as admin.
2. Add 16 works by the same publisher.
3. View the publisher's page via `/publishers/<Publisher>`.
4. Make sure that the count of works matches the number of covers present in the carousel.
5. View the publisher's work in search via `/search?q=publisher%3A"<Publisher>"&mode=everything`.
6. Make sure that the count of works matches the number of works rendered.

### Evidence
Look at the page buttons in these images.  I do not have any covers for these works.

Page1:
page1

Page2:
page2

Page3:
page3

### CODE REVIEW

**Comment by cbolgiano** [comment]
Here is my second attempt to solve this issue.  If there is a better way please let me know or push a commit.

File: openlibrary/templates/books/custom_carousel.html [hunk 1]
```diff
@@ -63,7 +63,7 @@ <h2 class="home-h2"><a href="$url">$title</a></h2>
         $if load_more:
           $ load_more_config = {
           $  "url": load_more["url"].replace("&amp;", "&"),
-          $  "pageMode": load_more.get("mode", "offset"),
+          $  "pageMode": load_more.get("mode", "page"),
```
Comment (jdlrobson) [inline openlibrary/templates/books/custom_carousel.html]: . I'm not sure if `offset` is used on another page with a carousel.. can you confirm confidently (I can't :-)) ? If not, parameter to the macro would be safer.

Comment (jdlrobson) [reply openlibrary/templates/books/custom_carousel.html]: I think if we can keep offset.. even better.

Comment (cbolgiano) [reply openlibrary/templates/books/custom_carousel.html]: Oh nice!

**Comment by cbolgiano** [comment]
That was the ticket jdlrobson, thank you.

**Comment by jdlrobson** [comment]
Thanks! While testing this I discovered #2501 but that has nothing to do with this patch. I like how simple your solution ended up being. Thanks for investigating this issue!

**Pull request merged**